build_tools/print_driver_gpu_info.py

```diff
@@ -2,316 +2,130 @@
-        gpu_name = gpu_name_match.group(1).strip() if gpu_name_match else None
-
-        # GFX target from ISA line: amdgcn-amd-amdhsa--gfx942
-        isa_match = re.search(r"amdgcn-amd-amdhsa--(gfx[0-9a-zA-Z]+)", block)
-        if isa_match:
-            gpu_target = isa_match.group(1).strip()
-        else:
-            # Fallback: "Name: gfx942" inside the GPU block
-            name_match = re.search(
-                r"^\s*Name:\s*(gfx[0-9a-zA-Z]+)", block, re.MULTILINE
-            )
-            gpu_target = name_match.group(1).strip() if name_match else None
-
-        gpu_agents.append(
-            {
-                "agent_index": agent_index,
-                "gpu_name": gpu_name,
-                "gpu_target": gpu_target,
-            }
-        )
-
-    if gpu_agents:
-        info["gpu_agents"] = gpu_agents
-        # Representative gpu_target for backwards compatibility
-        first_target = gpu_agents[0].get("gpu_target")
-        if first_target:
-            info["gpu_target"] = first_target
-
-    return info
-
-
-def collect_info() -> Dict[str, Any]:
-    """Run tools and collect structured info."""
-    info: Dict[str, Any] = {}
-
-    # Prefer SDK-installed binaries in ./build/bin (your local + CI pattern),
-    # then fall back to PATH.
-    amd_smi_candidates = [
-        "./build/bin/amd-smi",
-        "amd-smi",
-    ]
-    rocminfo_candidates = [
-        "./build/bin/rocminfo",
-        "rocminfo",
-    ]
-
-    # amd-smi / amdsmi
-    amd_path, amd_ok, amd_out = run_tool_with_candidates("amd-smi", amd_smi_candidates)
-    if amd_path:
-        info["amd_smi_path"] = amd_path
-    info["amd_smi_available"] = amd_ok
-    if amd_ok:
-        info.update(parse_amd_smi(amd_out))
-    else:
-        info["amd_smi_error"] = (amd_out or "").strip()
-
-    # rocminfo
-    rocminfo_path, roc_ok, roc_out = run_tool_with_candidates(
-        "rocminfo", rocminfo_candidates
-    )
-    if rocminfo_path:
-        info["rocminfo_path"] = rocminfo_path
-    info["rocminfo_available"] = roc_ok
-    if roc_ok:
-        info.update(parse_rocminfo(roc_out))
-    else:
-        info["rocminfo_error"] = (roc_out or "").strip()
-
-    return info
-
-
-def print_human_readable(info: Dict[str, object]) -> None:
-    """Emit log for CI."""
+    for cmd in candidates:
+        exe = cmd[0]
+
+        resolved = shutil.which(exe)
+        if resolved:
+            cmd_to_run = [resolved] + cmd[1:]
+            log(f"\n=== {label} ===")
+            run_command(cmd_to_run)
+            return
+
+        resolved_path = Path(exe)
+        if resolved_path.exists():
+            cmd_to_run = [str(resolved_path)] + cmd[1:]
+            log(f"\n=== {label} ===")
+            run_command(cmd_to_run)
+            return
+
+    # Nothing found
+    exe_name = candidates[0][0] if candidates else "<unknown>"
+    log(f"\n=== {label} ===")
+    log(f"{exe_name}: command not found")
```

Comment:
nit: the `candidates: List[List[str]]` typing here makes code like this confusing to read (what does `[0][0]` correspond to?). We could make this more explicit:

```python
rocminfo_candidates = [
    {
        "command": str(bin_dir / "amd-smi"),
        "args": ["static"],
    },
    {
        "command": "amd-smi",
        "args": ["static"],
    },
]
run_candidates("amd-smi static", amd_smi_candidates)
```

or more concise by separating out the command, the args, and the search paths ("candidates"):

```python
run_command(
    command="amd-smi",
    args=["static"],
    extra_command_search_paths=[bin_dir],
)
```

It doesn't matter _that_ much here given how short the script is now, but I like the second approach.

Reply:
Simplified this with the second approach.